delta_poker/env_wrapper.py

```diff
@@ -0,0 +1,176 @@
+import time
+from typing import Callable, Dict, Optional
+
+import numpy as np
+from gym.spaces import Box, Discrete
+from pettingzoo.utils import BaseWrapper
+
+
+class DeltaEnv(BaseWrapper):
+    def __init__(
+        self,
+        env,
+        opponent_choose_move: Callable,
+        verbose: bool = False,
+        render: bool = False,
+        game_speed_multiplier: int = 0,
+    ):
+
+        super().__init__(env)
+
+        self.opponent_choose_move = opponent_choose_move
+        self.render = render
+        self.verbose = verbose
+        self.game_speed_multiplier = game_speed_multiplier
+
+        # The player to move first is randomised by the env
+        self.player_agent = "player_0"
+        self.opponent_agent = "player_1"
+
+        self.most_recent_move: Dict[str, Optional[int]] = {
+            self.player_agent: None,
+            self.opponent_agent: None,
+        }
+
+        # TODO: Generalise to different games
+        self.action_space = Discrete(4)
+        self.observation_space = Box(low=0, high=1, shape=(72,))
```

Comment:
Is there a new version of this class that does generalise to other games? Presume so, since it was used for Go?

If so, does it make sense for this file to live in `delta_utils`, rather than here?

Reply:
https://learney.atlassian.net/browse/LRN-858?atlOrigin=eyJpIjoiMGYxYzUzNmUwOWNhNGUxYzk5NWE2OWRhNmNlY2M3ODgiLCJwIjoiaiJ9